gradle/libs.versions.toml

```diff
@@ -10,11 +10,11 @@ opentelemetryProto = "1.3.1-alpha"
 
 # otel agent, we rely on the '*-alpha' and get the non-alpha dependencies transitively
 # updated from upstream agent with .ci/update-upstream.sh
-opentelemetryJavaagentAlpha = "2.3.0-alpha"
+opentelemetryJavaagentAlpha = "2.4.0-alpha"
```

Comment:
[for reviewer] updating to the latest version allows to also bump the latest SDK and align with the version used for contrib (maybe a further improvement would be to ensure that we don't have any mismatch).